src/v2/providers/alerts/billing.ts

```diff
@@ -0,0 +1,105 @@
+import { getEndpointAnnotation, FirebaseAlertData } from '.';
+import { CloudEvent, CloudFunction } from '../../core';
+import * as options from '../../options';
+
+/**
+ * The internal payload object for billing plan updates.
+ * Payload is wrapped inside a FirebaseAlertData object.
+ */
+export interface PlanUpdatePayload {
+  ['@type']: 'com.google.firebase.firebasealerts.PlanUpdatePayload';
+  billingPlan: string;
+  principalEmail: string;
+}
+
+/**
+ * The internal payload object for billing plan automated updates.
+ * Payload is wrapped inside a FirebaseAlertData object.
+ */
+export interface PlanAutomatedUpdatePayload {
+  ['@type']: 'com.google.firebase.firebasealerts.PlanAutomatedUpdatePayload';
+  billingPlan: string;
+}
+
+interface WithAlertType {
+  alertType: string;
+}
+/**
+ * A custom CloudEvent for billing Firebase Alerts (with custom extension attributes).
+ */
+export type BillingEvent<T> = CloudEvent<FirebaseAlertData<T>, WithAlertType>;
+
+/** @internal */
+export const planUpdateAlert = 'billing.planUpdate';
+/** @internal */
+export const automatedPlanUpdateAlert = 'billing.automatedPlanUpdate';
+
+/** @internal */
+type BillingEventHandler<T> = (event: BillingEvent<T>) => any | Promise<any>;
```

Comment:
This typedef makes code complete worse (it asks for a BillingEventHandler instead of a function signature)

Reply:
Oh shoot, we should probably change it back for better usability then, right?

Reply:
I wanted to test out the automcomplete and see what shows, but I couldn't get to any of the billing handlers at all.

@colerogers Can you double check if this is working for you?

1) `npm run build:pack`
2) Install the generated tgz locally on your function
3) Try to get import to work.

Reply:
In fact, I couldn't even get `lib/v2/providers/billing.js` to show up. what gives?

Reply:
@taeold I was able to get it to work, just pushed an update. Seems like we can't rely on the package.json export path alone, we need to also export from the root index.ts. I think there was a comment about exporting on one of the other PRs and I ended up changing it on all of them

Reply:
@inlined Removed BillingEventHandler type from the latest commit